tests/unit/test_help.py

```diff
@@ -96,10 +97,28 @@ def test_pager_with_args(self):
     def test_no_groff_exists(self):
         renderer = FakePosixHelpRenderer()
         renderer.exists_on_path['groff'] = False
-        with self.assertRaisesRegexp(ExecutableNotFoundError,
-                                     'Could not find executable named "groff"'):
+        expected_error = 'Could not find executable named "groff"'
+        with self.assertRaisesRegexp(ExecutableNotFoundError, expected_error):
             renderer.render('foo')
 
+    @skip_if_windows('Requires POSIX system.')
+    def test_no_pager_exists(self):
+        fake_pager = 'foobar'
+        os.environ['MANPAGER'] = fake_pager
+        renderer = FakePosixHelpRenderer()
+        renderer.exists_on_path[fake_pager] = False
+
+        renderer.exists_on_path['groff'] = True
+        renderer.mock_popen.communicate.return_value = (b'foo', '')
+
+        stdout = six.StringIO()
+        captured_stdout = sys.stdout
+        sys.stdout = stdout
```

Comment:
This will clobber sys.stdout on failure.

In general:

* Try not to patch anything
* If you must patch, do it with mock.patch using either a decorator, context manager, or setup/teardown methods.

Strong preference on option 1.